sympy/physics/control/tests/test_lti.py

```diff
@@ -485,6 +529,538 @@ def test_TransferFunction_is_biproper():
+    assert DiscreteTransferFunction.from_rational_expression(expr_5, s) == \
+        DiscreteTransferFunction((2 + 3*s)*(5 + 2*s), (9 + 3*s)*(5 + 2*s**2), s)
+    assert DiscreteTransferFunction.from_rational_expression(expr_6, s) == \
+        DiscreteTransferFunction((9*s**4 + 4*s**2 + 8), (s + 1)*(s + 9), s)
+    assert H1 == H2
+    assert DiscreteTransferFunction.from_rational_expression(expr_8, s,
+                                                       sampling_time = 0.3) == \
+        DiscreteTransferFunction(2*s**2 + 3*s + 2, s**2 + 1, s, 0.3)
+
+    # classmethod from_coeff_lists
+    tf1 = DiscreteTransferFunction.from_coeff_lists([1, 2], [3, 4, 5], s)
+    num2 = [p**2, 2*p]
+    den2 = [p**3, p + 1, 4]
+    tf2 = DiscreteTransferFunction.from_coeff_lists(num2, den2, s)
+    num3 = [1, 2, 3]
+    den3 = [0, 0]
+
+    assert tf1 == DiscreteTransferFunction(s + 2, 3*s**2 + 4*s + 5, s)
+    assert tf2 == DiscreteTransferFunction(p**2*s + 2*p, p**3*s**2 + s*(p + 1) + 4, s)
+    raises(ZeroDivisionError, lambda:
+           DiscreteTransferFunction.from_coeff_lists(num3, den3, s))
+
+    # classmethod from_zpk
+    zeros = [4]
+    poles = [-1+2j, -1-2j]
+    gain = 3
+    tf1 = DiscreteTransferFunction.from_zpk(zeros, poles, gain, s, 3)
+
+    assert tf1 == DiscreteTransferFunction(3*s - 12,
+                                     (s + 1.0 - 2.0*I)*(s + 1.0 + 2.0*I), s, 3)
+
+    # explicitly cancel poles and zeros.
+    tf0 = DiscreteTransferFunction(s**5 + s**3 + s, s - s**2, s)
+    a = DiscreteTransferFunction(-(s**4 + s**2 + 1), s - 1, s)
+    assert tf0.simplify() == simplify(tf0) == a
+
+    tf1 = DiscreteTransferFunction((p + 3)*(p - 1), (p - 1)*(p + 5), p)
+    b = DiscreteTransferFunction(p + 3, p + 5, p)
+    assert tf1.simplify() == simplify(tf1) == b
+
+    # expand the numerator and the denominator.
+    G1 = DiscreteTransferFunction((1 - s)**2, (s**2 + 1)**2, s)
+    G2 = DiscreteTransferFunction(1, -3, p)
+    c = (a2*s**p + a1*s**s + a0*p**p)*(p**s + s**p)
+    d = (b0*s**s + b1*p**s)*(b2*s*p + p**p)
+    e = a0*p**p*p**s + a0*p**p*s**p + a1*p**s*s**s + a1*s**p*s**s + \
+        a2*p**s*s**p + a2*s**(2*p)
+    f = b0*b2*p*s*s**s + b0*p**p*s**s + b1*b2*p*p**s*s + b1*p**p*p**s
+    g = a1*a2*s*s**p + a1*p*s + a2*b1*p*s*s**p + b1*p**2*s
+    G3 = DiscreteTransferFunction(c, d, s)
+    G4 = DiscreteTransferFunction(a0*s**s - b0*p**p, (a1*s + b1*s*p)*(a2*s**p + p), p,
+                             2.4)
+
+    assert G1.expand() == DiscreteTransferFunction(s**2 - 2*s + 1, s**4 + 2*s**2 + 1,
+                                             s)
+    assert tf1.expand() == DiscreteTransferFunction(p**2 + 2*p - 3, p**2 + 4*p - 5, p)
+    assert G2.expand() == G2
+    assert G3.expand() == DiscreteTransferFunction(e, f, s)
+    assert G4.expand() == DiscreteTransferFunction(a0*s**s - b0*p**p, g, p, 2.4)
+    assert G4.expand() != DiscreteTransferFunction(a0*s**s - b0*p**p, g, p)
+
+    # purely symbolic polynomials.
```

Comment:
Change comment to "testing that subs works", I assumed it was testing the construction of a purely symbolic transfer function but maybe you do that somewhere else.